internal/weaver/remoteweavelet.go

```diff
@@ -286,6 +288,26 @@ func NewRemoteWeavelet(ctx context.Context, regs []*codegen.Registration, bootst
 		return nil
 	})
 
+	// Start a signal handler to detect when the process is killed. This isn't
+	// perfect, as we can't catch a SIGKILL, but it's good in the common case.
+	done := make(chan os.Signal, 1)
+	signal.Notify(done, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		<-done
+		for _, c := range w.componentsByName {
+			if !c.implReady.Load() {
+				continue
+			}
+			// Call Shutdown method if available.
+			if i, ok := c.impl.(interface{ Shutdown(context.Context) error }); ok {
```

Comment:
Hmm, I would prefer to keep it as `Shutdown`. In the next PR, I will add update our docs to talk about `Shutdown` and the guarantees it provides. Otherwise, if some day we'll provide other guarantees we may have to rename/add a new method, which feels a bit weird to me.

Reply:
I added the documentation as well.